ThirdPartyAdapters/unity/unity/src/main/java/com/google/ads/mediation/unity/UnityAdapter.java

```diff
@@ -215,96 +220,71 @@ public void showInterstitial() {
+    public void onUnityAdsStart(String placementId) {
+        // Unity Ads video ad started playing. Google Mobile Ads SDK does not support
+        // callbacks for Interstitial ads when they start playing.
     }
 
     @Override
-    public void onBannerLoaded(BannerView bannerView) {
-        Log.v(TAG, "Unity Ads Banner finished loading banner for placement: " + mBannerView.getPlacementId());
-        bannerListener.onAdLoaded(UnityAdapter.this);
+    public void onUnityAdsClick(String s) {
+        // Unity Ads ad clicked.
+        if (mMediationInterstitialListener != null) {
+            mMediationInterstitialListener.onAdClicked(UnityAdapter.this);
+            // Unity Ads doesn't provide a "leaving application" event, so assuming that the
+            // user is leaving the application when a click is received, forwarding an on ad
+            // left application event.
+            mMediationInterstitialListener.onAdLeftApplication(UnityAdapter.this);
+        }
     }
 
     @Override
-    public void onBannerClick(BannerView bannerView) {
-        bannerListener.onAdClicked(UnityAdapter.this);
+    public void onUnityAdsPlacementStateChanged(String placementId,
+                                                UnityAds.PlacementState oldState,
+                                                UnityAds.PlacementState newState) {
+        // This callback is not forwarded to the adapter and the
+        // adapter should use the onUnityAdsReady and onUnityAdsError callbacks to forward
+        // Unity Ads SDK state to Google Mobile Ads SDK.
     }
 
     @Override
-    public void onBannerFailedToLoad(BannerView bannerView, BannerErrorInfo bannerErrorInfo) {
-        Log.w(TAG, "Unity Ads Banner encountered an error: " + bannerErrorInfo.errorMessage);
-        if (bannerErrorInfo.errorCode == BannerErrorCode.NO_FILL) {
-            bannerListener.onAdFailedToLoad(UnityAdapter.this, AdRequest.ERROR_CODE_NO_FILL);
-        } else {
-            bannerListener.onAdFailedToLoad(UnityAdapter.this, AdRequest.ERROR_CODE_INTERNAL_ERROR);
+    public void onUnityAdsFinish(String placementId, UnityAds.FinishState finishState) {
+        // Unity Ads ad closed.
+        if (mMediationInterstitialListener != null) {
+            mMediationInterstitialListener.onAdClosed(UnityAdapter.this);
         }
     }
 
     @Override
-    public void onBannerLeftApplication(BannerView bannerView) {
-        bannerListener.onAdLeftApplication(UnityAdapter.this);
+    public void onUnityAdsError(UnityAds.UnityAdsError unityAdsError, String placementId) {
+        // Send Ad Failed to load event.
+        if (placementId.equals(getPlacementId()) && mMediationInterstitialListener != null) {
+            Log.e(TAG, "Failed to load Interstitial ad from Unity Ads: " +
+                    unityAdsError.toString());
+            mMediationInterstitialListener.onAdFailedToLoad(UnityAdapter.this,
+                    AdRequest.ERROR_CODE_NO_FILL);
+        }
```

Comment:
the only case when `onUnityAdsError ` is called, only when we have show error.

Reply:
Is this the correct fix?
if (placementId.equals(getPlacementId()))
        {
            Log.e(TAG, "Failed to show Interstitial ad from Unity Ads: " + unityAdsError.toString());
        }